torchinductor/ir.py

```diff
@@ -758,8 +758,11 @@ def create(cls, x, *, dim=None):
             )
             return ReinterpretView(storage, new_layout)
 
-        # redirect to a generic view
-        return View.create(x, [s for s in x.get_size() if s != 1])
+        if dim is None:
+            # redirect to a generic view
+            return View.create(x, [s for s in x.get_size() if s != 1])
+        else:
+            return View.create(x, [s for i, s in enumerate(x.get_size()) if i != dim])
```

Comment:
need also to assert that that size(dim) is 1?

Reply:
Thanks I'll add that.